### PR TITLE
MODINVSTOR-486: Create view in Inventory for OAI-PMH

### PR DESCRIPTION
**Pull request by mrdpopov**
# Purpose

Since OAI-PMH response is going to be expanded with some holdings and items field, database view should be created in Inventory to retrieve this data in a more convenient way.

### CODE REVIEW

**Comment by MikhailFokanov** [comment]
@marc-johnson, could you please have a look on this PR, if it should be reviewed by your team.

**Comment by marcjohnson-kint** [comment]
@MikhailFokanov 

> could you please have a look on this PR, if it should be reviewed by your team.

Thank you for reaching out to me. @mrdpopov also did so on Slack, and I responded there.

Under the new development process, I don't believe there is any requirement for the Core Functional team (which is very low on backend capacity at the moment) to review this pull request.

@craigmcnally please correct me if I am wrong.

I will, however, be writing a comment on the JIRA with some follow up design questions.

(Please use my work github account, @marcjohnson-kint for notifications, I'm aware it is confusing that I have two)

File: src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                                            coalesce(
+                                                    (instance.jsonb ->> 'discoverySuppress')::bool,
+                                                    false))
+                                    group by 1
+     )
+
+select instanceIdsAndDatesInRange.instanceId,
+       instanceIdsAndDatesInRange.maxDate,
+       false as deleted,
+       (select to_jsonb(itemAndHoldingsAttrs) as instanceFields
```
Comment (craigmcnally) [inline src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql:114]: is instanceFields a misleading name here?  Doesn't this include data from location/holding/item records?

File: src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    from audit_holdings_record audit_holdings_record
+             join audit_item audit_item
+                  on (audit_item.jsonb ->> '{record,holdingsRecordId}')::uuid =
+                     audit_holdings_record.id
+    where ((audit_holdings_record.jsonb #>>
+            '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+        or
+           (audit_item.jsonb #>> '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2))
+),
+     instanceIdsAndDatesInRange as (SELECT instanceId,
```
Comment (craigmcnally) [inline src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql:96]: inconsistent use of case... SELECT vs select

**Comment by craigmcnally** [comment]
the inconsistent indentation makes this difficult to follow at times.  Please consider making it more consistent throughout.

File: src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql (hunk 2 above)
Comment (mrdpopov) [reply src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql:96]: Corrected, now all keywords are in lower-case.

**Comment by mrdpopov** [comment]
@craigmcnally I have reformatted the script with IDE's general formatter. Does it look better?

**Comment by mrdpopov** [comment]
In OAI-PMH response feed those are counted as instance fields indeed.

**Pull request merged**